Course/Course/Presentation/Container/CourseContainerViewModel.swift

```diff
@@ -119,7 +120,44 @@ public class CourseContainerViewModel: BaseCourseViewModel {
             }
         }
     }
+    
+    @MainActor
+    func getCourseDeadlineInfo(courseID: String, withProgress: Bool = true) async {
+        isShowProgress = withProgress
+        do {
+            let courseDeadlineInfo = try await interactor.getCourseDeadlineInfo(courseID: courseID)
+            withAnimation {
+                self.courseDeadlineInfo = courseDeadlineInfo
+            }
+            isShowProgress = false
+        } catch let error {
+            isShowProgress = false
+            if error.isInternetError || error is NoCachedDataError {
+                errorMessage = CoreLocalization.Error.slowOrNoInternetConnection
+            } else {
+                errorMessage = CoreLocalization.Error.unknownError
+            }
+        }
+    }
 
+    @MainActor
+    func shiftDueDates(courseID: String, withProgress: Bool = true) async {
+        isShowProgress = withProgress
+        dueDatesShifted = false
```

Comment:
Do we really need to set it false, as it will be false by default and even after dates shifting.